### PR TITLE
#2975 implement new fee select component

### LINKED ISSUE #160: [9.5] Implement new FeeSelect component #2975
https://github.com/bitshares/bitshares-ui/issues/2975

### PR DESCRIPTION
**Pull request by VerevkinAlexander**
<h2>General</h2>
Closes #160 

I have added FeeAsset component to the next modals
app/components/Dashboard/SimpleDepositWithdraw.jsx
app/components/Showcases/Barter.jsx
app/components/Modal/HtlcModal.jsx
app/components/Modal/WithdrawModalNew.jsx
app/components/Modal/DirectDebitModal.jsx
app/components/Modal/DirectDebitClaimModal.jsx
also I have deleted old unused code and performed refactoring

<h2>General</h2>
Please make sure the following is done:

- Pull request is onto develop
- If you haven't already, have a look at
  - https://github.com/bitshares/bitshares-ui/blob/develop/CODE_OF_CONDUCT.md
  - https://github.com/bitshares/bitshares-ui/blob/develop/CONTRIBUTING.md

<h2>Code Preparation</h2>

_Please review all your changes one last time before committing_

- [ ] Check for unused code
- [ ] No unrelated changes are included
- [ ] None of the changed files are reformatting only
- [ ] Code is self explanatory or documented
- [ ] All written text is properly translated (english language)

<h2>Testing</h2>

_The branch has been tested on the following browsers (desktop and mobile view)_

- [ ] Chrome 
- [ ] Opera
- [ ] Firefox
- [ ] Safari

<h2>User interface changes</h2>

_Delete this section if there weren't any UI changes. Please make sure you tested your changes in all themes_

- [ ] Dark
- [ ] Light
- [ ] Midnight

image
image
image
image
image
image

### CODE REVIEW

**Comment by sschiessl-bcp** [comment]
When I open 
http://localhost:8080/settings/general
with a hard refresh I get
image

Please check

**Comment by VerevkinAlexander** [comment]
@sschiessl-bcp,
I was not able to reproduce this issue even after a hard refresh or clearing of localStorage.

**Comment by sschiessl-bcp** [comment]
Ok thanks. I could fix it in another branch

File: app/components/Dashboard/SimpleDepositWithdraw.jsx [hunk 1]
```diff
@@ -56,26 +55,20 @@ class DepositWithdrawContent extends DecimalChecker {
             fee_asset_id:
                 ChainStore.assets_by_symbol.get(props.fee_asset_symbol) ||
                 "1.3.0",
-            feeStatus: {},
+            feeAmount: new Asset({amount: 0}),
```
Comment (sschiessl-bcp) [inline app/components/Dashboard/SimpleDepositWithdraw.jsx]: Please use only feeAsset, which contains itself amount and asset_id

File: app/components/Dashboard/SimpleDepositWithdraw.jsx [hunk 2]
```diff
@@ -222,77 +215,9 @@ class DepositWithdrawContent extends DecimalChecker {
...
-                },
-                this._checkFeeStatus
-            );
-        });
-    }
-
     _getCurrentBalance() {
         let balances = this.props.balance
-            ? [this.props.balance]
+            ? [ChainStore.getObject(this.props.balance)]
```
Comment (sschiessl-bcp) [inline app/components/Dashboard/SimpleDepositWithdraw.jsx:225]: Why was this addeD?

File: app/components/Dashboard/SimpleDepositWithdraw.jsx [hunk 3]
```diff
@@ -499,68 +400,26 @@ class DepositWithdrawContent extends DecimalChecker {
...
+                    <FeeAssetSelector
+                        label="showcases.barter.fee_when_proposal_executes"
+                        account={this.props.account}
+                        trxInfo={{
+                            type: "transfer",
+                            options: ["price_per_kbyte"],
+                            data: {
+                                type: "memo",
+                                content:
+                                    this.props.backingCoinType.toLowerCase() +
```
Comment (sschiessl-bcp) [inline app/components/Dashboard/SimpleDepositWithdraw.jsx]: Refactor to avoid code duplication

File: app/components/Modal/HtlcModal.jsx [hunk 4]
```diff
@@ -968,8 +825,8 @@ class HtlcModal extends React.Component {
             operation && operation.type === "create"
                 ? counterpart.translate("showcases.htlc.create_htlc")
                 : isExtend
-                    ? counterpart.translate("showcases.htlc.extend_htlc")
-                    : counterpart.translate("showcases.htlc.redeem_htlc");
+                ? counterpart.translate("showcases.htlc.extend_htlc")
```
Comment (sschiessl-bcp) [inline app/components/Modal/HtlcModal.jsx:828]: This refactoring here is not reproduced by my prettifier hook. Are you sure you don't have a local format on save or something? It's really weird

File: app/components/Modal/HtlcModal.jsx [hunk 5]
```diff
@@ -1140,42 +997,21 @@ class HtlcModal extends React.Component {
...
-                                                display_balance={balance_fee}
-                                                // tabIndex={tabIndex++}
-                                                error={
-                                                    this.state
-                                                        .hasPoolBalance ===
-                                                    false
-                                                        ? "transfer.errors.insufficient"
-                                                        : null
+                                        <FeeAssetSelector
+                                            label="transfer.fee"
```
Comment (sschiessl-bcp) [inline app/components/Modal/HtlcModal.jsx]: label="transfer.fee" is the default, remove across the whole PR is this value is used

File: app/components/Showcases/Barter.jsx [hunk 6]
```diff
@@ -78,13 +72,18 @@ export default class Barter extends Component {
                 to_barter: [{message: "", shown: false}],
                 escrow: [{message: "", shown: false}]
             },
-            proposal_fee: 0,
+            proposal_fee: {
```
Comment (sschiessl-bcp) [inline app/components/Showcases/Barter.jsx:75]: Why is this split in amount and real amount necessary?

**Review by sschiessl-bcp — CHANGES_REQUESTED** [review]

File: app/components/Dashboard/SimpleDepositWithdraw.jsx (hunk 2 above)
Comment (VerevkinAlexander) [reply app/components/Dashboard/SimpleDepositWithdraw.jsx:225]: Because there is was an issue, the balance props contain the asset id only and  the balances props contains the scope of the assets

File: app/components/Modal/HtlcModal.jsx (hunk 4 above)
Comment (VerevkinAlexander) [reply app/components/Modal/HtlcModal.jsx:828]: really weird, agreed. Maybe VScode and Webshtorm have different default settings?

File: app/components/Modal/HtlcModal.jsx (hunk 5 above)
Comment (VerevkinAlexander) [reply app/components/Modal/HtlcModal.jsx]: done

File: app/components/Dashboard/SimpleDepositWithdraw.jsx (hunk 3 above)
Comment (VerevkinAlexander) [reply app/components/Dashboard/SimpleDepositWithdraw.jsx]: done

File: app/components/Showcases/Barter.jsx (hunk 6 above)
Comment (VerevkinAlexander) [reply app/components/Showcases/Barter.jsx:75]: fixed

File: app/components/Dashboard/SimpleDepositWithdraw.jsx (hunk 1 above)
Comment (VerevkinAlexander) [reply app/components/Dashboard/SimpleDepositWithdraw.jsx]: done

**Comment by VerevkinAlexander** [comment]
plz review

**Comment by sschiessl-bcp** [comment]
Forwarded
